ros2_batch_job/__main__.py

```diff
@@ -273,7 +283,56 @@ def get_args(sysargv=None):
     return args
 
 
-def process_coverage(args, job):
+def include_python_coverage_in_report(args, package=None):
+    # no arguments means report all packages in buildspace
+    root_directory_to_report = args.buildspace
+    if package:
+        root_directory_to_report = os.path.join(root_directory_to_report, package)
+
+    coverage_xml_files = glob.glob(
+        os.path.join(root_directory_to_report, '**', 'coverage.xml'),
+        recursive=True)
+    for coverage_xml_path in coverage_xml_files:
+        # python coverage detected: move the coverage.xml file to buildspace to be reported
+        dest_file_to_report = os.path.join(args.buildspace, coverage_xml_path.replace(os.path.sep, '.'))
```

Comment:
Can you share an example name for `dest_file_to_report`? Or point to a test build with these change to look at the resulting workspace.

Reply:
My expectation is to have file paths like: `build/my_python_foo_pkg.subdir1.coverage.xml` . The goal was to make them unique and store all them in a single dir (build/) so the cobertura plugin does not need to look into the whole filesystem to find them. Let me launch a testing run to demonstrate that this is actually true,

Reply:
This is one the files generated https://ci.ros2.org/job/test_linux_coverage/ws/ws/build/build.domain_coordinator.coverage.xml. The patch is visible in the URL.

Reply:
The referenced resource isn't available anymore.